api/explore/explore.js

```diff
@@ -0,0 +1,123 @@
+const express = require('express');
+const asyncHandler = require('express-async-handler');
+const crypto = require('crypto');
+const holiday = require('./next_holiday_calc');
+const MonthJson = require('./ingredients_of_the_month.json');
+const BoxesJson = require('./random_boxes.json');
+const recs = require('../recommend.js');
+const search = require('../search.js');
+
+
+const router = express.Router();
+
+function randomChoice(arr, del) {
+  const ind = Math.floor(arr.length * Math.random());
+  const retval = arr[ind];
+  if (del) {
+    // To avoid repetitions, the arrays will get smaller as we go.
+    arr.splice(ind, 1);
+  }
+  return retval;
+}
+
+async function innerSearch(searchString, amount, request) {
+  const bool = {
+    must: {
+      simple_query_string: {
+        query: searchString,
+        fields: ['title', 'ingredients', 'tags'],
+      },
+    },
+  };
+
+  await search.searchFunc(bool, 0, amount, request)
+    .then((values) => values)
+    .catch((err) => {
+      throw (err);
+    });
+}
+
+async function GetBoxes(size, dateString, request, amount) {
+  const monthIngs = MonthJson[new Date().toLocaleDateString('default', { month: 'long' })];
+  const keys = Object.keys(BoxesJson);
+  let n = size;
+  keys.push('ingredient');
+  const retval = {};
+  const nextHoliday = holiday(dateString);
+
+  if (nextHoliday) {
+    const holidayRecipes = await innerSearch(holiday(dateString), amount, request);
+    retval['next holiday'] = { name: nextHoliday, recipes: holidayRecipes };
+    n -= 1;
+  }
+
+  while (n > 0) {
```

Comment:
Don't use a while loop, there is literally no need to use these in JS, use maps and what not